lib/src/solid/utils/key_management.dart

```diff
@@ -653,3 +657,96 @@ class _PrvKeyRecord {
   /// The corresponding decrypted private key
   String? key;
 }
+
+/// [RecipientPubKey] is a class to store public keys of another POD.
+/// This public key is used to share encrypted data to this POD
+
+class RecipientPubKey {
+  /// Constructor
+  RecipientPubKey({required this.recipientWebId});
+
+  /// The webId of the recipient
+  String recipientWebId;
+
+  /// The content of the public key
+  String? _recipientPubKeyContent;
+
+  /// The public key with prefix and suffix
+  RSAPublicKey? _recipientPubKey;
+
+  /// Get the public key
+  Future<RSAPublicKey> getPubKey() async {
+    if (_recipientPubKey == null) {
+      await _setPubKey();
+    }
+
+    return _recipientPubKey!;
+  }
+
+  /// Get the public key content
+  Future<String> getPubKeyContent() async {
+    if (_recipientPubKeyContent == null) {
+      await _setPubKey();
+    }
+
+    return _recipientPubKeyContent!;
+  }
+
+  /// Set the public key
+  Future<void> _setPubKey() async {
+    /// Get recipient's public key
+    final recipientPubKeyUrl =
+        recipientWebId.replaceAll(profCard, await getPubKeyPath());
+
+    // Get and parse the pubKeyFile
+    final map = await loadPrvTTL(recipientPubKeyUrl);
+
+    if (!map.containsKey(recipientPubKeyUrl)) {
+      throw Exception('Invalid content in file: "$recipientPubKeyUrl"');
+    }
+
+    _recipientPubKeyContent = map[recipientPubKeyUrl][pubKeyPred] as String;
+
+    final recipientPubKeyStr = genPubKeyStr(_recipientPubKeyContent as String);
+
+    final parser = RSAKeyParser();
+    _recipientPubKey = parser.parse(recipientPubKeyStr) as RSAPublicKey;
+  }
+
+  /// Encrypt a given value using public key
+  Future<String> encryptData(String dataVal) async {
+    if (_recipientPubKey == null) {
+      await _setPubKey();
+    }
+
+    final encrypter = Encrypter(RSA(publicKey: _recipientPubKey));
+    return encrypter.encrypt(dataVal).base64;
+  }
+}
+
+/// Generate TTL string from triples stored in a map:
+/// {subject: {predicate: object}}
+/// where
+/// - subject: the URL of a file
+/// - predicate-object: the key-value pairs to be stores in the file
+
+String _genTTLStr(Map<String, Map<String, String>> tripleMap) {
```

Comment:
The function body of `_genTTLStr()` looks identical to that of `tripleMapToTTStr()` in `src/solid/utils/rdf.dart`.